### PR TITLE
Log to `Console` and `Debug` when targeting MAUI 

### PR DESCRIPTION
**Pull request by jamescrosswell**
Resolves https://github.com/getsentry/sentry-dotnet/issues/3341

# Approach

There are a couple of different ways we could solve this. One would be a [CompositeLogger](https://github.com/getsentry/sentry-dotnet/blob/bdbadd6712ce87ce591c2b54c2d8b090e49c269f/src/Sentry/Infrastructure/CompositeLogger.cs) which could be initialised like this:
```csharp
                DiagnosticLogger = new CompositeLogger(
                    new ConsoleDiagnosticLogger(DiagnosticLevel),
                    new TraceDiagnosticLogger(DiagnosticLevel)
                );
```

That would basically log to both. It's a bit of a hack since technically, you could instantiate each of the composite loggers with a different `DiagnosticLevel`. That could be guarded against by checking all the levels match in the constructor but it's not as elegant as I'd like. 

Alternatively, we could default to a [DebugDiagnosticLogger](https://github.com/getsentry/sentry-dotnet/blob/bdbadd6712ce87ce591c2b54c2d8b090e49c269f/src/Sentry/Infrastructure/DebugDiagnosticLogger.cs) which uses `Debug.WriteLine`. That works in both Rider and Visual Studio. 

It's a small change to the behavriour of the API. Anyone who wanted to retain the previous behaviour could explicitly set `SentryOptions.DiagnosticLogger` to be a `ConsoleDiagnosticLogger`.

In this PR, I've gone with that option.

### CODE REVIEW

File: integration-test/runtime.Tests.ps1 [hunk 1]
```diff
@@ -13,12 +13,14 @@ Describe 'Console app NativeAOT (<framework>)' -ForEach @(
...
 using Sentry.Extensibility;
 using Sentry.Protocol.Envelopes;
+using Sentry.Infrastructure;
 
 // Initialize the Sentry SDK.  (It is not necessary to dispose it.)
 SentrySdk.Init(options =>
 {
     options.Dsn = args[0];
     options.Debug = true;
+    options.DiagnosticLogger = new ConsoleDiagnosticLogger(SentryLevel.Debug);
```
Comment (bitsandfoxes) [inline integration-test/runtime.Tests.ps1:23]:
```suggestion
    // Some tests rely on the logs being on the console
    options.DiagnosticLogger = new ConsoleDiagnosticLogger(SentryLevel.Debug);
```

**Comment by bitsandfoxes** [comment]
I did a quick check with our `Blazor.WASM` sample and the logs in the browser seem to rely on the console. Ideally, we'd have some fallback to the console logger on platforms where we know it will go into the void.

**Comment by bitsandfoxes** [comment]
I'm fine with the behaviour change in general because I assume that this is nothing that affects users in production
It enables debug functionality where it previously did not work.

**Comment by jamescrosswell** [comment]
> I did a quick check with our `Blazor.WASM` sample and the logs in the browser seem to rely on the console.

Nice catch... tricky though. As we initialise Blazor using plain old `SentrySdk.Init`, I can't think of an easy way to determine, when initialising the SDK, if the app is a Blazor WASM app or not. 

We could create a WASM integration with a`WebAssemblyHostBuilder.UseSentry` extension to initialise the SDK for WASM applications and set `SentryOptions.DiagnosticLogger = new ConsoleDiagnosticLogger(SentryLevel.Debug)` in that case. Given the popularity of Blazor/WASM, it might be worth it to make Sentry as usable as possible in this scenario. We could use that extension method to encapsulate some of the other things involved in correctly configuring the SDK for use with WASM as well.

Alternatively, we could just document/show in the WASM sample that the DiagnosticLogger needs to be set to something non-default (explaining that `Debug.WriteLine` doesn't work in WASM applications).

**Comment by bitsandfoxes** [comment]
I tried using this all day and I think I changed my mind. I found this to be fairly irritating to have the SDK logs show up somewhere else than the rest of the program's logs.
And when running samples from the command line there were just no logs at all. So I'd have to manually set the `ConsoleDiagnosticLogger` in the options.
TLDR; this felt like a major breaking change.

**Comment by jamescrosswell** [comment]
> TLDR; this felt like a major breaking change.

We could revisit the initial solution but implement it more simply, like this:

```csharp
public class TraceAndConsoleDiagnosticLogger : DiagnosticLogger
{
    public TraceAndConsoleDiagnosticLogger(SentryLevel minimalLevel) : base(minimalLevel)
    {
    }

    protected override void LogMessage(string message)
    {
        Trace.WriteLine(message);
        Console.WriteLine(message);
    }
}
```

It would create duplication if you had a console tracer configured, but I'd say that's a bit of an edge case (and we could guide people to change the default logger in those cases).

**Comment by bitsandfoxes** [comment]
I think we agreed on the sync to try to log to both via the MAUI setup to catch those VS users?

**Review by bitsandfoxes — CHANGES_REQUESTED** [review]

File: src/Sentry.Maui/SentryMauiAppBuilderExtensions.cs [hunk 2]
```diff
@@ -49,7 +50,16 @@ public static MauiAppBuilder UseSentry(this MauiAppBuilder builder, string dsn)
...
+            void MauiOptions(SentryMauiOptions o)
+            {
+                configureOptions(o);
+                if (o is { Debug: true, DiagnosticLogger: null })
+                {
+                    o.DiagnosticLogger = new ConsoleAndTraceDiagnosticLogger(o.DiagnosticLevel);
+                }
+            }
+
+            services.Configure((Action<SentryMauiOptions>)MauiOptions);
```
[the remark this answers is not in the document]
Comment (jamescrosswell) [reply src/Sentry.Maui/SentryMauiAppBuilderExtensions.cs:62]: That actually sounded like a really good idea. When I went to implement it however:

a) I spotted this:

https://github.com/getsentry/sentry-dotnet/blob/bd62cf0421cc7a5d7b7804f8461fae00d47d43b8/src/Sentry.Maui/Internal/SentryMauiOptionsSetup.cs#L30-L31

Good, good... so onto the constructor. 
```csharp
        DiagnosticLogger = new ConsoleAndTraceDiagnosticLogger(DiagnosticLevel);
```

The problem with that is that it assumes the `DiagnosticLevel` won't change. If the users sets the `DiagnosticLevel `to anything other than the default, this wouldn't be reflected in the `DiagnosticLogger` that we setup (before they configured that level). 

So I think all of this has to be delayed until where it's being done currently (once we have all the bits of the puzzle to put together).

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,8 +2,9 @@
 
 ## Unreleased
 
-### Fixes
+## Fixes
```
Comment (bitsandfoxes) [inline CHANGELOG.md:5]:
```suggestion
### Fixes
```

File: src/Sentry.Maui/SentryMauiAppBuilderExtensions.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (jamescrosswell) [reply src/Sentry.Maui/SentryMauiAppBuilderExtensions.cs]: No idea what my brain was thinking in the earlier comment. Moving to SentryMauiOptionsSetup seems to work fine.

**Review by bitsandfoxes — APPROVED** [review]
Awesome! Thanks!
